media-gfx/zbar/zbar-0.23.ebuild

```diff
@@ -0,0 +1,187 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# TODO: drop python2_7
+# * https://github.com/mchehab/zbar/issues/27#issuecomment-491463811
+# * https://bugs.gentoo.org/694800
+PYTHON_COMPAT=( python2_7 python3_{6,7,8} )
+
+inherit autotools flag-o-matic java-pkg-opt-2 multilib-minimal python-single-r1 virtualx
+
+DESCRIPTION="Library and tools for reading barcodes from images or video"
+HOMEPAGE="https://github.com/mchehab/zbar"
+SRC_URI="https://linuxtv.org/downloads/${PN}/${P}.tar.gz"
+
+KEYWORDS="~amd64 ~arm ~x86"
+LICENSE="LGPL-2.1"
+SLOT="0"
+
+IUSE="dbus graphicsmagick gtk gtk2 imagemagick introspection java jpeg png python qt5 static-libs test +threads v4l X xv"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	introspection? ( || ( gtk gtk2 ) )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	test? ( ${PYTHON_REQUIRED_USE}
+		X png || ( graphicsmagick imagemagick ) )
+	qt5? ( X )"
```

Comment:
Is that actually true?

Reply:
yeah. https://github.com/mchehab/zbar/blob/1d51925e6cc9151f4a73781989f21fcd7b57ef32/configure.ac#L641

but `/usr/lib64/libzbarqt.so.0.1.2` library can build without Xorg libraries if I use the patch for fixing `#include <QX11Info>`

I'm not sure that good idea

Reply:
no, that's showing me an option, which is appropriately described by the dependencies below. not a requirement.

If you have to patch the path to `QX11Info` to make it work without it that indicates a missing `ifdef `in the sources.

Reply:
> https://github.com/mchehab/zbar/blob/1d51925e6cc9151f4a73781989f21fcd7b57ef32/configure.ac#L641

it was actually for compatibility between qt4 and qt5

I can use this patch for fix it and (it works fine without 'X' flag):
```diff
diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2019-05-23 23:37:12.000000000 +0300
+++ b/configure.ac	2020-01-22 12:27:14.190755156 +0300
@@ -638,13 +638,9 @@
 
 AC_ARG_VAR([MOC], [full path to Qt moc program])
 
-AS_IF([test "x$have_x" = "xyes"],
-      [qt_extra="Qt5X11Extras >= 5.0"],
-      [qt_extra=""])
-
 AS_IF([test "x$with_qt" != "xno"],
       [PKG_CHECK_MODULES([QT],
-		         [Qt5Core >= 5 Qt5Gui >= 5 Qt5Widgets >= 5.0 $qt_extra],,
+		         [Qt5Core >= 5 Qt5Gui >= 5 Qt5Widgets >= 5.0 Qt5X11Extras >= 5.0],,
 		         [with_qt5 = "no"
 		          PKG_CHECK_MODULES([QT],
 					    [QtCore >= 4 QtGui >= 4],,

```

Reply:
>Don't search for Qt5X11Extras on systems that do not have X, like MinGW 

https://github.com/mchehab/zbar/commit/3aa61a2b84cd7c1732481176e2094e3f3b00df54#diff-67e997bcfdac55191033d57a16d1408a